reset-pass/account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/controller/MainController.java

```diff
@@ -0,0 +1,1283 @@
+            model.addAttribute("warnMsg", messageSource.getMessage("validate.userNotDefined", new String[]{uiService}, userLocale));
+            return "error.html";
+        }
+
+        //user who added this person
+        User addedBy = u.getCreatedBy();
+        
+        //Do not display sites and other welcome page information if user is updating userid
+        if(va.getAccountStatus().equals(ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE)) {
+            String submitUpdate = messageSource.getMessage("submit.update", null, userLocale);
+            model.addAttribute("accountTitle", submitUpdate);
+            model.addAttribute("eid", va.getEid());
+            model.addAttribute("submitBtn", submitUpdate);
+
+        } else {
+            model.addAttribute("accountTitle", messageSource.getMessage("activateAccount.title", null, userLocale));
+            model.addAttribute("eid", u.getDisplayId());
+            model.addAttribute("wait1", messageSource.getMessage("validate.wait.newUser.1", null, userLocale));
+            String linkText = messageSource.getMessage("validate.wait.newUser.2", new String[]{uiService}, userLocale);
+            String transferMembershipsURL = getViewURL("transferMemberships", va);
+            String labelLink = "<a href="+transferMembershipsURL+">"+linkText+"</a>";
+            model.addAttribute("wait2", labelLink);
+            model.addAttribute("addDetailsSub", messageSource.getMessage("submit.new.account", null, userLocale));
+            model.addAttribute("submitBtn", messageSource.getMessage("submit.new.account", null, userLocale));
+            //we need to know which sites they're a member of:
+            Set<String> groups = authzGroupService.getAuthzGroupsIsAllowed(EntityReference.getIdFromRef(va.getUserId()), "site.visit", null);
+            Iterator<String> git = groups.iterator();
+            List existingSites = new ArrayList<>();
+            while (git.hasNext()) {
+
+                String groupRef = git.next();
+                String groupId = EntityReference.getIdFromRef(groupRef);
+                if (!existingSites.contains(groupId)) {
+                    log.debug("groupId is " + groupId);
+
+                    try {
+                        Site s = siteService.getSite(groupId);
+                        model.addAttribute("siteListItem", s.getTitle());
+
+                        existingSites.add(groupId);
+                    
+                    } catch (IdUnusedException e) {
+
+                        log.error(e.getMessage(), e);
+                    }
+                }
+            }
+
+            if (existingSites.size() >= 2)  {
+                model.addAttribute("welcome", messageSource.getMessage("validate.welcome.plural", new String[]{uiService}, userLocale));
+
+            } else {
+                model.addAttribute("welcome", messageSource.getMessage("validate.welcome.single", new String[]{uiService}, userLocale));
```

Comment:
You can use one line here:

model.addAttribute("welcome", messageSource.getMessage(existingSites.size() >= 2 ? "validate.welcome.plural" : "validate.welcome.single", new String[]{uiService}, userLocale));